localstack/ext/java/src/main/java/cloud/localstack/LocalstackTestRunner.java

```diff
@@ -241,21 +301,84 @@ public void run() {
 		}
 	}
 
+	/**
+	 * for Windows there are a several key differnces:
+	 *   1.  the command passed in to bash needs to be surrounded by quotes
+	 *   2.  windows automatically appends the system path onto the linux path when invoking bash
+	 *   3.  windows file paths need to be converted to work in linux (i.e. change '\' to '/' and 'c:' to '/mnt/c')
+	 *   4.  Process.waitFor() never returns when the java process invokes bash and thus waitFor(timeout) is being used
+	 */
+	private static Process execWindows(boolean wait, String ... cmd) {
+		try {
+			int timeout = 15;
+			if (cmd.length == 1 && !new File(cmd[0]).exists()) {
+				String commandArg = "\"" + cmd[0] + "; exit\"";
+				timeout = getProcessTimeout(commandArg);
+				LOG.info(String.format("executing (timeout = %ds) : bash -c %s", timeout, commandArg));
+				cmd = new String[]{"bash", "-c", commandArg};
+			}
+			ProcessBuilder builder = new ProcessBuilder(cmd);
+			builder.environment().put(ENV_LOCALSTACK_PROCESS_GROUP, ENV_LOCALSTACK_PROCESS_GROUP);
+
+			final Process p = builder.start();
+
+			if (wait) {
+				if (!p.waitFor(timeout, TimeUnit.SECONDS)) {
+					LOG.info("process timeout, assume completed successfully...");
+					Runtime.getRuntime().addShutdownHook(new Thread() {
+						public void run() {
+							killProcess(p);
+						}
+					});
+				}
+			} else {
+				/* make sure we destroy the process on JVM shutdown */
+				Runtime.getRuntime().addShutdownHook(new Thread() {
+					public void run() {
+						killProcess(p);
+					}
+				});
+			}
+			return p;
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 *  these numbers are based on empirical observations with some padding added in
+	 */
+	private static int getProcessTimeout(String cmd) {
+		int result = 15;
+
+		if (cmd.contains("make ") && cmd.contains(" infra")) {
+			result = 120;
+		} else if (cmd.contains("make install")) {
+			result = 300;
+		} else if (cmd.contains("git clone ")) {
+			result = 10;
+		}
+
+		return result;
+	}
+
 	private void setupInfrastructure() {
 		synchronized (INFRA_STARTED) {
 			// make sure everything is installed locally
 			ensureInstallation();
 			// make sure we avoid any errors related to locally generated SSL certificates
 			TestUtils.disableSslCertChecking();
 
-			if(INFRA_STARTED.get() != null) return;
-			String[] cmd = new String[]{"make", "-C", TMP_INSTALL_DIR, "infra"};
+			// check to see if the server process is already running or created
+			if (INFRA_STARTED.get() != null || isAlive()) return;
+
 			Process proc;
 			try {
-				proc = exec(false, cmd);
+				LOG.info("starting...");
```

Comment:
Perhaps a more "verbose"/detailed message like:
```
LOG.info("Starting LocalStack infrastructure services from installation directory: " + getTmpInstallDir());
```
Then we can also remove line 381.